src/map/status.cpp

```diff
@@ -13960,6 +13966,18 @@ TIMER_FUNC(status_change_timer){
 			return 0;
 		}
 		break;
+		
+	case SC_GRADUAL_GRAVITY:
+		if (--(sce->val4) >= 0) {
+			int hp = status->max_hp * sce->val2 / 100;
+
+			if (!status_charge(bl, hp, 0))
+				status_zap(bl, hp, 0);
+			if (sc->getSCE(type))
+				sc_timer_next(interval + tick);
+			return 0;
```

Comment:
```suggestion
```
This part will be handled at the bottom of the function.
Looks for `// If status has an interval and there is at least 100ms remaining time, wait for next interval` to find it.